packages/fscomponents/src/components/Form/Templates/above/LabelAboveStatefulTextbox.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from 'react';
+import { Image, Text, TextInput, View } from 'react-native';
+import {
+  defaultTextboxStyle,
+  errorIcon,
+  getColor,
+  StatefulTextboxProps,
+  StatefulTextboxState,
+  successIcon
+} from '../formBoilerplate';
+
+export default class LabelAboveTextbox extends Component<StatefulTextboxProps,
+StatefulTextboxState> {
+
+  state: StatefulTextboxState = {
+    active: false,
+    validated: false
+  };
+
+  onFocus = () => {
+    this.setState({
+      active: true,
+      validated: false
+    });
+  }
+
+  onBlur = () => {
+    this.setState({
+      active: false,
+      validated: true
+    });
+  }
+
+  render(): JSX.Element {
+    const {locals} = this.props;
+
+    const prevonBlur = locals.onBlur;
+    const prevOnFocus = locals.onFocus;
+
+    locals.onBlur = () => {
+      this.onBlur();
+      if (prevonBlur instanceof Function) {
+        prevonBlur();
+      }
+    };
+
+    locals.onFocus = () => {
+      this.onFocus();
+      if (prevOnFocus instanceof Function) {
+        prevOnFocus();
+      }
+    };
+
+    locals.placeholder = '';
+
+    const defaultStyle = defaultTextboxStyle(locals);
+
+    const {
+      alertStyle,
+      checkStyle,
+      controlLabelStyle,
+      activeErrorField,
+      error,
+      formGroupStyle,
+      help,
+      rightTextboxIconStyle,
+      textboxFullBorderStyle,
+      textboxViewStyle
+    } = defaultStyle;
+
+    const getIcon = () => {
+      return (this.props.locals.hasError ?
+      <Image source={errorIcon} style={alertStyle}/> :
+      <Image source={successIcon} style={checkStyle}/>
+      );
+    };
+
+    const color = getColor(this.state, locals);
+
+    return (
+      <View>
+        <View style={formGroupStyle}>
+          <View>
+            <Text style={[controlLabelStyle, {color}]}>{locals.label}</Text>
+          </View>
+          <View style={textboxViewStyle}>
+            <TextInput
+              accessibilityLabel={locals.label}
+              /* tslint:disable-next-line */
```

Comment:
Hey do you think it is better to define `private input: RefObject<TextInput>;` 
and then initialize it in constructor like this `this.input = React.createRef<TextInput>();` 
and use it like this `ref={this.input}`?